spec/exe/floe_spec.rb

```diff
@@ -1,14 +1,122 @@
+    expect(lines).to start_with("Usage: floe [options] workflow input [workflow2 input2]")
+    expect(lines).to include("v#{Floe::VERSION}")
 
     # it should also include options from runners
-    expect(output).to include("Container runner options:")
+    expect(lines).to include("Container runner options:")
+  end
+
+  it "with no parameters" do
+    result = AwesomeSpawn.run(exe)
+    expect(result).to be_failure
+
+    lines = result.error.lines(:chomp => true)
+    expect(lines.first).to eq("Error: workflow/input pairs must be specified.")
+  end
+
+  it "with a bare workflow and no input" do
+    output = AwesomeSpawn.run!(exe, :params => [workflow]).output
+    lines = output.lines(:chomp => true)
+
+    expect(lines.first).to include("checking 1 workflows...")
+    expect(lines.last).to eq("{}")
+  end
+
+  it "with a bare workflow and input" do
+    output = AwesomeSpawn.run!(exe, :params => [workflow, '{"foo":1}']).output
+    lines = output.lines(:chomp => true)
+
+    expect(lines.first).to include("checking 1 workflows...")
+    expect(lines.last).to eq('{"foo"=>1}')
+  end
+
+  it "with a bare workflow and --input" do
+    output = AwesomeSpawn.run!(exe, :params => [workflow, {:input => '{"foo":1}'}]).output
+    lines = output.lines(:chomp => true)
+
+    expect(lines.first).to include("checking 1 workflows...")
+    expect(lines.last).to eq('{"foo"=>1}')
+  end
+
+  it "with --workflow and no input" do
+    output = AwesomeSpawn.run!(exe, :params => {:workflow => workflow}).output
+    lines = output.lines(:chomp => true)
+
+    expect(lines.first).to include("checking 1 workflows...")
+    expect(lines.last).to eq("{}")
+  end
+
+  it "with --workflow and --input" do
+    output = AwesomeSpawn.run!(exe, :params => {:workflow => workflow, :input => '{"foo":1}'}).output
+    lines = output.lines(:chomp => true)
+
+    expect(lines.first).to include("checking 1 workflows...")
+    expect(lines.last).to eq('{"foo"=>1}')
+  end
+
+  it "with a bare workflow and --workflow" do
+    result = AwesomeSpawn.run(exe, :params => [workflow, {:workflow => workflow}])
+    expect(result).to be_failure
+
+    lines = result.error.lines(:chomp => true)
+    expect(lines.first).to eq("Error: cannot specify both --workflow and bare workflows.")
+  end
+
+  it "with --input but no workflows" do
+    result = AwesomeSpawn.run(exe, :params => {:input => '{"foo":1}'})
+    expect(result).to be_failure
+
+    lines = result.error.lines(:chomp => true)
+    expect(lines.first).to eq("Error: workflow(s) must be specified.")
+  end
+
+  it "with multiple bare workflow/input pairs" do
+    output = AwesomeSpawn.run!(exe, :params => [workflow, '{"foo":1}', workflow, '{"foo":2}']).output
+    lines = output.lines(:chomp => true)
+
+    expect(lines.first).to include("checking 2 workflows...")
+    expect(lines.last(7).join("\n")).to eq(<<~OUTPUT.chomp)
+      workflow
+      ===
+      {"foo"=>1}
+
+      workflow
+      ===
+      {"foo"=>2}
```

Comment:
This was surprising to me (here and of course everywhere else), as I assumed the output would be JSON formatted and not Ruby formatted.  So, the question is, should it be JSON?  If so, it's an easy fix for a followup PR.

Reply:
I also expected either the workflow.asl file name or perhaps the workflow comment (or both), so just the word "workflow" was also surprising.

Reply:
I modified this to run multiple workflows.
Goals:

- run multiple providers.
- run in a way similar to how the provider runs
- allow easy migration to the non-polling code.
- easily determine that all workflows were run and their output status.

Modifying it sounds great.

Reply:
I wasn't really thinking about post processing, but that would be useful in the single workflow case.  Even so, it was more of an expectation thing...for some reason I expected a json output.